app/controllers/users.js

```diff
@@ -22,5 +24,34 @@ module.exports = {
         logger.error(`DB: ${error.errors[0]}`);
         res.status(500).json({ error: error.errors[0].message });
       });
+  },
+
+  userNewSession(req, res, next) {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      logger.error('User input invalid.');
+      logger.error(errors.array());
+      return res.status(422).json({ errors: errors.array() });
+    }
+    const { email, password } = req.body;
+    User.findOne({ where: { email } })
+      .then(user => {
+        if (!user) {
+          logger.error('Email is not registered.');
+          return res.status(401).json({ error: 'User auth failed. Check your email or password.' });
+        }
+        if (bcrypt.compareSync(password, user.password)) {
+          logger.info(`User ${email} authenticated.`);
+          const token = jwt.sign(JSON.stringify(user), config.secret);
```

Comment:
Be careful that you are adding all the user instance in the jwt, incluiding the hashed password, and the jwt is only Base64 encoded so that info will be public.

Reply:
Is it not encrypted using the secret hash?

Reply:
Fixed